src/models/sankey.js

```diff
@@ -0,0 +1,343 @@
+nv.models.sankey = function() {
+    'use strict';
+
+    // Sources:
+    // - https://bost.ocks.org/mike/sankey/
+    // - https://github.com/soxofaan/d3-plugin-captain-sankey
+
+    //============================================================
+    // Public Variables with Default Settings
+    //------------------------------------------------------------
+
+    var sankey = {},
+        nodeWidth = 24,
+        nodePadding = 8,
+        size = [1, 1],
+        nodes = [],
+        links = [],
+        sinksRight = true;
+
+    sankey.nodeWidth = function(_) {
+        if (!arguments.length) return nodeWidth;
+        nodeWidth = +_;
+        return sankey;
+    };
```

Comment:
@liquidpele Refactored using the `_options`but not sure if that's the right way to pass and call functions, I haven't seen any kind of similar "callback" function to start the rendering for a chart.
See the last 3 properties: https://github.com/ksgy/nvd3/blob/master/src/models/sankey.js#L309-L311